tests/test_qdrant_client.py

```diff
@@ -257,6 +259,57 @@ def test_point_upload(prefer_grpc):
     assert result_count.count > 100
 
 
+@pytest.mark.parametrize("prefer_grpc", [False, True])
+@pytest.mark.parametrize("parallel", [1, 2])
+def test_upload_collection(prefer_grpc, parallel):
+    size = 3
+    batch_size = 2
+    client = QdrantClient(prefer_grpc=prefer_grpc, timeout=TIMEOUT)
+
+    client.recreate_collection(
+        collection_name=COLLECTION_NAME,
+        vectors_config=VectorParams(size=size, distance=Distance.DOT),
+        timeout=TIMEOUT,
+    )
+    vectors = [
+        [1.0, 2.0, 3.0],
+        [4.0, 5.0, 6.0],
+        [7.0, 8.0, 9.0],
+        [10.0, 11.0, 12.0],
+        [13.0, 14.0, 15.0],
+    ]
```

Comment:
Right now the behavior is to stop at the shortest iterator of any of `ids`, `vectors`, or `ids`. Is it possible to emit a warning when this happens? E.g, for when it stopped with any of those un-exhausted

Reply:
Yeah, we can consider it as a separate issue

Reply:
I've just realised that having iterators of different length is a valid scenario, e.g. it is valid when ids iterator is infinite.

We can only check the number of ids/payloads/vectors right before making a request, however this check won't help when the smallest iterator is divisible by batch_size